extensions/arc/src/extension.ts

```diff
@@ -0,0 +1,478 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as vscode from 'vscode';
+import * as azdata from 'azdata';
+import { IconPathHelper, cssStyles } from './constants';
+import { BasicAuth } from './controller/auth';
+import { DatabaseRouterApi, DuskyObjectModelsDatabaseService, DuskyObjectModelsDatabase, DuskyObjectModelsDatabaseServiceArcPayload } from './controller/generated/dusky/api';
+import { EndpointsRouterApi, EndpointModel, RegistrationRouterApi, RegistrationResponse, TokenRouterApi, TokenModel } from './controller/generated/v1/api';
+
+// Controller information
+const controllerUrl = 'https://0.0.0.0:30080';
+const auth = new BasicAuth('username', 'password');
+
+export async function activate(context: vscode.ExtensionContext): Promise<void> {
+	IconPathHelper.setExtensionContext(context);
+	const dashboard: azdata.window.ModelViewDashboard = azdata.window.createModelViewDashboard('Azure Arc - Postgres');
+	dashboard.registerTabs(async (view: azdata.ModelView) => {
+		// TODO: Loading icon while we fetch information
+
+		// Database service information
+		const dbName = 'my-postgres2';
+		const dbNamespace = 'default';
+
+		// Get database service from the controller
+		const duskyApi = new DatabaseRouterApi(auth.username, auth.password, controllerUrl);
+		duskyApi.setDefaultAuthentication(auth);
+		const servicePromise: Promise<{ body: DuskyObjectModelsDatabaseService }> = duskyApi.getDuskyDatabaseService(dbNamespace, dbName);
+
+		// Get controller endpoints
+		const endpointApi = new EndpointsRouterApi(auth.username, auth.password, controllerUrl);
```

Comment:
You're probably going to want to move all this kind of logic into some sort of model class that handles fetching the data so you can easily refresh it and keep that logic separate from the UI. 

See https://github.com/microsoft/azuredatastudio/blob/master/extensions/big-data-cluster/src/bigDataCluster/dialog/bdcDashboardModel.ts for an example